samcli/commands/init/init_templates.py

```diff
@@ -102,6 +106,37 @@ def clone_templates_repo(self):
                 if expected_previous_clone_local_path.exists():
                     self._git_repo.local_path = expected_previous_clone_local_path
 
+    def _check_upsert_templates(self, shared_dir: Path, cloned_folder_name: str) -> bool:
+        """
+        Check if the app templates repository should be cloned, or if cloning should be skipped.
+
+        Parameters
+        ----------
+        shared_dir: Path
+            Folder containing the aws-sam-cli shared data
+
+        cloned_folder_name: str
+            Name of the directory into which the app templates will be copied
+
+        Returns
+        -------
+        bool
+            True if the cache should be updated, False otherwise
+
+        """
+        cache_dir = shared_dir / cloned_folder_name
```

Comment:
Should we use `Path` or `os.path` to join the paths according to the different operating systems?

Reply:
if `shared_dir` and `cloned_folder_name` are `Path` object, the `/` operator should support all platforms already.

Reply:
Thanks @hawflau, I didn't know this before. Looking at typing, it is a `Path` object, so it should work as expected then.

Reply:
`Pathlib` overloads the `/` operator so it should handle paths correctly. This might not be the most read-able solution though so I'll make it more explicit.